### PR TITLE
Improvement: Updated the serviceaccount flag for multiple subjects.

### LINKED ISSUE #108608: kubectl create help implies a list of service accounts can be references in a clusterrolebinding
### What happened?

I'm not sure whether this is the documentation being unclear, or a bug with kubectl itself but I suspect the former.

Given this help page
```bash
$ kubectl create clusterrolebinding --help
Create a cluster role binding for a particular cluster role.
# Omitted for brevity
      --serviceaccount=[]: Service accounts to bind to the clusterrole, in the format <namespace>:<name>
```
I expected to be able to use
```bash
$ kubectl create clusterrolebinding pipelineSAs --clusterrole=view --serviceaccount=["ns1:pipeline", "ns2:pipeline"]
```
Instead I got the following error
```bash
$ kubectl create clusterrolebinding pipelinesas --clusterrole=view --serviceaccount=["ns1:pipeline", "ns2:pipeline"]
error: exactly one NAME is required, got 2
See 'kubectl create clusterrolebinding -h' for help and examples
```

### What did you expect to happen?

To get the same output as defining the flag twice

aka
```bash
$ kubectl create clusterrolebinding pipelinesas --clusterrole=view --serviceaccount=ns1:pipeline --serviceaccount=ns2:pipeline
clusterrolebinding.rbac.authorization.k8s.io/pipelinesas created
```
and

```bash
$ kubectl create clusterrolebinding pipelineSAs --clusterrole=view --serviceaccount=["ns1:pipeline", "ns2:pipeline"]
clusterrolebinding.rbac.authorization.k8s.io/pipelinesas created
```
should give the same output, or the help page shouldn't list `[]` as the parameter type for --serviceaccount

### How can we reproduce it (as minimally and precisely as possible)?

```bash
kubectl create ns ns1 
kubectl create ns ns2
kubectl --namespace ns1 create sa pipeline
kubectl --namespace ns2 create sa pipeline
kubectl create clusterrolebinding pipelineSAs --clusterrole=view --serviceaccount=["ns1:pipeline", "ns2:pipeline"]
```

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:41:01Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:34:54Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
N/A
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
$ uname -a
Linux controlplane 5.4.0-88-generic #99-Ubuntu SMP Thu Sep 23 17:29:00 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

# On Windows:
N/A
```

</details>

### Install tools

<details>
```console
$ kubectl -n kube-system get cm kubeadm-config  -oyaml
apiVersion: v1
data:
  ClusterConfiguration: |
    apiServer:
      extraArgs:
        authorization-mode: Node,RBAC
      timeoutForControlPlane: 4m0s
    apiVersion: kubeadm.k8s.io/v1beta3
    certificatesDir: /etc/kubernetes/pki
    clusterName: kubernetes
    controllerManager: {}
    dns: {}
    etcd:
      local:
        dataDir: /var/lib/etcd
    imageRepository: k8s.gcr.io
    kind: ClusterConfiguration
    kubernetesVersion: v1.23.1
    networking:
      dnsDomain: cluster.local
      podSubnet: 192.168.0.0/16
      serviceSubnet: 10.96.0.0/12
    scheduler: {}
kind: ConfigMap
metadata:
  name: kubeadm-config
  namespace: kube-system
```
</details>

### Container runtime (CRI) and version (if applicable)

<details>
      architecture: amd64
      bootID: 09b2e311-c6dc-40bb-9485-64c98f77cf66
      containerRuntimeVersion: containerd://1.5.5
      kernelVersion: 5.4.0-88-generic
      kubeProxyVersion: v1.23.1
      kubeletVersion: v1.23.1
</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by Shubham82**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:
This PR  updated the serviceaccont flag for multiple subjects under  `clusterrolebinding`  and `rolebinding`.

#### Which issue(s) this PR fixes:

Fixes #108608

#### Does this PR introduce a user-facing change?

```
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by Shubham82** [comment]
Hi @dougsland @eddiezane 
Could you please take look!

**Comment by RichardoC** [comment]
LGTM, thanks for the rapid fix!

**Comment by Shubham82** [comment]
/assign @brianpursley

PTAL!

**Review by brianpursley — CHANGES_REQUESTED** [review]
Wouldn't the same apply to the `--user` and `--group` flags?  If so, I'm thinking they should say the same thing.

Also, I'm not sure if the term "subject" makes sense here, as that terminology is not used anywhere else in the help text.  Maybe just "service accounts" instead, to be consistent.

**Comment by Shubham82** [comment]
Thanks, @brianpursley for taking look at this.

> Wouldn't the same apply to the `--user` and `--group` flags? If so, I'm thinking they should say the same thing.

Yes, we can add the same thing for `--user` and` --group` flags.  

> Also, I'm not sure if the term "subject" makes sense here, as that terminology is not used anywhere else in the help text. Maybe just "service accounts" instead, to be consistent.

Adding `service accounts` instead of `subject` make sense.

I will update it.

**Comment by Shubham82** [comment]
@brianpursley, The --user flag is not shown under the options when we run `kubectl create clusterrolebinding --help`.

```
$ kubectl create clusterrolebinding --help

Options:
      --allow-missing-template-keys=true: If true, ignore any errors in templates when a field or map key is missing in
the template. Only applies to golang and jsonpath output formats.
      --clusterrole='': ClusterRole this ClusterRoleBinding should reference
      --dry-run='none': Must be "none", "server", or "client". If client strategy, only print the object that would be
sent, without sending it. If server strategy, submit server-side request without persisting the resource.
      --field-manager='kubectl-create': Name of the manager used to track field ownership.
      --group=[]: Groups to bind to the clusterrole
  -o, --output='': Output format. One of:
json|yaml|name|go-template|go-template-file|template|templatefile|jsonpath|jsonpath-as-json|jsonpath-file.
      --save-config=false: If true, the configuration of current object will be saved in its annotation. Otherwise, the
annotation will be unchanged. This flag is useful when you want to perform kubectl apply on this object in the future.
      --serviceaccount=[]: Service accounts to bind to the clusterrole, in the format <namespace>:<name>
      --show-managed-fields=false: If true, keep the managedFields when printing objects in JSON or YAML format.
      --template='': Template string or path to template file to use when -o=go-template, -o=go-template-file. The
template format is golang templates [http://golang.org/pkg/text/template/#pkg-overview].
      --validate=true: If true, use a schema to validate the input before sending it 
```

This is the same for rolebinding also. 

could you please tell me why it is not shown or if it is a bug(i think it is) then can you point me to code where I can do changes?

**Comment by eddiezane** [comment]
The user flag here is shadowed by a global flag. You can see this by running `kubectl options`.

https://github.com/kubernetes/kubernetes/blob/699aeb735ff34d81d79817bf613203ac58e6edc3/staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go#L362

I think we have encountered this before. Please search for shadowed flags.

**Comment by Shubham82** [comment]
Hi @brianpursley 
I have addressed the comment, PTAL!

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_clusterrolebinding.go [hunk 1]
```diff
@@ -100,9 +100,9 @@ func NewCmdCreateClusterRoleBinding(f cmdutil.Factory, ioStreams genericclioptio
 	cmdutil.AddDryRunFlag(cmd)
 	cmd.Flags().StringVar(&o.ClusterRole, "clusterrole", "", i18n.T("ClusterRole this ClusterRoleBinding should reference"))
 	cmd.MarkFlagRequired("clusterrole")
-	cmd.Flags().StringArrayVar(&o.Users, "user", o.Users, "Usernames to bind to the clusterrole")
-	cmd.Flags().StringArrayVar(&o.Groups, "group", o.Groups, "Groups to bind to the clusterrole")
-	cmd.Flags().StringArrayVar(&o.ServiceAccounts, "serviceaccount", o.ServiceAccounts, "Service accounts to bind to the clusterrole, in the format <namespace>:<name>")
+	cmd.Flags().StringArrayVar(&o.Users, "user", o.Users, "Usernames to bind to the clusterrole. The flag can be repeated to add multiple users.")
+	cmd.Flags().StringArrayVar(&o.Groups, "group", o.Groups, "Groups to bind to the clusterrole. The flag can be repeated to add multiple groups.")
+	cmd.Flags().StringArrayVar(&o.ServiceAccounts, "serviceaccount", o.ServiceAccounts, "Service accounts to bind to the clusterrole, in the format <namespace>:<name>. The flag can be repeated to add multiple Service accounts.")
```
Comment (brianpursley) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_clusterrolebinding.go:105]:
```suggestion
	cmd.Flags().StringArrayVar(&o.ServiceAccounts, "serviceaccount", o.ServiceAccounts, "Service accounts to bind to the clusterrole, in the format <namespace>:<name>. The flag can be repeated to add multiple service accounts.")
```

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_rolebinding.go [hunk 2]
```diff
@@ -103,9 +103,9 @@ func NewCmdCreateRoleBinding(f cmdutil.Factory, ioStreams genericclioptions.IOSt
 	cmdutil.AddDryRunFlag(cmd)
 	cmd.Flags().StringVar(&o.ClusterRole, "clusterrole", "", i18n.T("ClusterRole this RoleBinding should reference"))
 	cmd.Flags().StringVar(&o.Role, "role", "", i18n.T("Role this RoleBinding should reference"))
-	cmd.Flags().StringArrayVar(&o.Users, "user", o.Users, "Usernames to bind to the role")
-	cmd.Flags().StringArrayVar(&o.Groups, "group", o.Groups, "Groups to bind to the role")
-	cmd.Flags().StringArrayVar(&o.ServiceAccounts, "serviceaccount", o.ServiceAccounts, "Service accounts to bind to the role, in the format <namespace>:<name>")
+	cmd.Flags().StringArrayVar(&o.Users, "user", o.Users, "Usernames to bind to the role. The flag can be repeated to add multiple users.")
+	cmd.Flags().StringArrayVar(&o.Groups, "group", o.Groups, "Groups to bind to the role. The flag can be repeated to add multiple groups.")
+	cmd.Flags().StringArrayVar(&o.ServiceAccounts, "serviceaccount", o.ServiceAccounts, "Service accounts to bind to the role, in the format <namespace>:<name>. The flag can be repeated to add multiple Service accounts.")
```
Comment (brianpursley) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_rolebinding.go:108]:
```suggestion
	cmd.Flags().StringArrayVar(&o.ServiceAccounts, "serviceaccount", o.ServiceAccounts, "Service accounts to bind to the role, in the format <namespace>:<name>. The flag can be repeated to add multiple service accounts.")
```

**Review by brianpursley — CHANGES_REQUESTED** [review]

**Comment by Shubham82** [comment]
Thanks, @brianpursley for taking look at this.
I will update it.

**Comment by Shubham82** [comment]
@brianpursley 
PTAL

**Review by brianpursley — APPROVED** [review]
/lgtm

**Pull request merged**